### PR TITLE
Make explicit that auth file is not supported anymore

### PR DESCRIPTION
**Pull request by lmazuel**
Fix https://github.com/Azure/azure-sdk-for-python/issues/16791

### CODE REVIEW

**Review by rakshith91 — APPROVED** [review]

File: sdk/core/azure-common/azure/common/client_factory.py [hunk 1]
```diff
@@ -152,6 +160,12 @@ def get_client_from_json_dict(client_class, config_dict, **kwargs):
     :param dict config_dict: A config dict.
     :return: An instantiated client
     """
+    if _is_autorest_v3(client_class):
+        raise ValueError(
+            "Auth file or JSON dict are deprecated auth approach and are not supported anymore. "
```
Comment (mccoyp) [inline sdk/core/azure-common/azure/common/client_factory.py:165]: This implies that these approaches are generally deprecated -- should the deprecation be specific to track 2 clients? e.g. by saying something like "azure-common factory methods are deprecated for `client_class.__name__` and are not supported anymore."

Comment (lmazuel) [reply sdk/core/azure-common/azure/common/client_factory.py:165]: It's under a "track2" `if` statement, is that not enough you think?

Comment (mccoyp) [reply sdk/core/azure-common/azure/common/client_factory.py:165]: I think it should cover the cases where this should be raised, but I just wonder if users would be led to think that these factory methods wouldn't work for non-deprecated track 1 clients as well. Does that make sense?

Comment (mccoyp) [reply sdk/core/azure-common/azure/common/client_factory.py:165]: This is assuming that most users wouldn't look at the source code to notice that the error is specific to this condition

Comment (lmazuel) [reply sdk/core/azure-common/azure/common/client_factory.py:165]: Well, I think we don't love auth file, and therefore being sligtly wide in the message is ok to me. But this is a good point to think.

**Review by mccoyp — APPROVED** [review]

**Review by chlowell — APPROVED** [review]

**Comment by lmazuel** [comment]
Seems the storage failure is not related to this PR, forcing merge

**Pull request merged**